hearth/lib/hearth/telemetry/span_kind.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Hearth
+  module Telemetry
+    # Type of span.
+    # Can be used to describe relationships between spans in a trace.
+    module SpanKind
```

Comment:
You can possibly get rid of this module and make these something like Hearth::Telemetry::INTERNAL_SPAN. This current style is like some sort of C enum design.

Reply:
[OpenTelemetry-Ruby implementation of SpanKind](https://github.com/open-telemetry/opentelemetry-ruby/blob/406cabe54f8fb4937d0f6b73fe830a51c04fdb92/api/lib/opentelemetry/trace/span_kind.rb#L4)

Since Observability is heavily based on OTel, I'll opt to leave these defined under SpanKind